packages/x-components/src/x-modules/history-queries/store/__tests__/actions.spec.ts

```diff
@@ -269,4 +269,88 @@ describe('testing history queries module actions', () => {
       expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(true);
     });
   });
+
+  describe('updateHistoryQueriesWithSearchResponse', () => {
+    const results: Result[] = [
+      { modelName: 'Result', id: '1' },
+      { modelName: 'Result', id: '2' }
+    ];
+    const totalResults = results.length;
+    let gato: HistoryQuery, perro: HistoryQuery;
+
+    beforeEach(() => {
+      [gato, perro] = createHistoryQueries('gato', 'perro');
+      resetStateWith({ historyQueries: [gato, perro] });
+    });
+
+    it('updates a history query if its response is a success', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1
+        },
+        status: 'success',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([{ ...gato, totalResults }, perro]);
+    });
+
+    it('does not update a history query if its response is an error', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1
+        },
+        status: 'error',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([gato, perro]);
+    });
+
+    it('does not update a history query if it is not stored', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'pato',
+          page: 1
+        },
+        status: 'success',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([gato, perro]);
+    });
+
+    it('updates a history query if it was made on a previous session', async () => {
+      gato.totalResults = 50;
+      gato.timestamp = store.state.sessionTimeStampInMs;
+      resetStateWith({ historyQueries: [gato, perro] });
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1
+        },
+        status: 'success',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([{ ...gato, totalResults }, perro]);
+    });
+
+    it('updates a history query if the new totalResults is higher', async () => {
+      gato.totalResults = 1;
```

Comment:
The test pass even if the new `totalResults` is lower, I guess it is due to both the HQ and session timestamps being 0 causing `isCurrentSessionHistoryQuery` in the action to be false. 

```suggestion
      gato.totalResults = 1;
      gato.timestamp = store.state.sessionTimeStampInMs + 100; // Make sure HQ is in current session
```

Could you also add a test to check that the HQ is not updated when the new `totalResults` is lower?